Libraries/LibChess/Chess.cpp

```diff
@@ -684,4 +687,40 @@ bool Board::operator==(const Board& other) const
     return turn() == other.turn();
 }
 
+void Board::set_resigned(Chess::Colour c)
+{
+    m_resigned = c;
+}
+
+String Board::result_to_string(Result r) const
+{
+    switch (r) {
+    case Result::CheckMate:
+        if (m_turn == Chess::Colour::White)
+            return "Black wins by Checkmate";
+        else
+            return "White wins by Checkmate";
```

Comment:
Don't use `else` after `return`.

Also, you should probably assert that `m_turn` is not `Chess::Colour::None`:
```cpp
ASSERT(m_turn != Chess::Colour::None);
return m_turn == Chess:Colour::White ? "Black wins by Checkmate" : "White wins by Checkmate";
```